File: PERL/Cap04/c04ex06.pl
# c04ex06.pl

#!/usr/bin/perl

use strict;
use warnings;

print "Entre o 1o. valor numerico: ";
my $A = <STDIN>;
chomp $A;

print "Entre o 2o. valor numerico: ";
my $B = <STDIN>;
chomp $B;

print "\n";
print "Escolha uma OPCAO de menu:\n";
print "\n";
print "[1] - Adicao\n";
print "[2] - Subtracao\n";
print "[3] - Multiplicacao\n";
print "[4] - Divisao\n";
print "\n";

print "==> ";
my $OPCAO = <STDIN>;
chomp $OPCAO;

print "\n";

if ($OPCAO == 1) {
  print "Calculo de adicao\n";
  print "Resultado: " . ($A + $B) . "\n";
} elsif ($OPCAO == 2) {
  print "Calculo de subtracao\n";
  print "Resultado: " . ($A - $B) . "\n";
} elsif ($OPCAO == 3) {
  print "Calculo de multiplicacao\n";
  print "Resultado: " . ($A * $B) . "\n";
} elsif ($OPCAO == 4) {
  print "Calculo de divisao\n";
  if ($B == 0) {
    print "Resultado: ERRO\n";
  } else {
    print "Resultado: " . ($A / $B) . "\n";
  }
} else {
  print "Entrada invalida\n";
  print "Tente novamente em outro momento\n";
}

print "\n";
print "Tecle <Enter> para encerrar... ";
<STDIN>;

File: RUBY/Cap04/c04ex06.rb
# c04ex06.rb

#!/usr/bin/env ruby

print "Entre o 1o. valor numerico: "
a = gets.chomp.to_f

print "Entre o 2o. valor numerico: "
b = gets.chomp.to_f

puts
puts "Escolha uma OPCAO de menu:"
puts
puts "[1] - Adicao"
puts "[2] - Subtracao"
puts "[3] - Multiplicacao"
puts "[4] - Divisao"
puts

print "==> "
opcao = gets.chomp.to_i

puts

case opcao
when 1
  puts "Calculo de adicao"
  puts "Resultado: #{a + b}"
when 2
  puts "Calculo de subtracao"
  puts "Resultado: #{a - b}"
when 3
  puts "Calculo de multiplicacao"
  puts "Resultado: #{a * b}"
when 4
  puts "Calculo de divisao"
  if b.zero?
    puts "Resultado: ERRO"
  else
    puts "Resultado: #{a / b}"
  end
else
  puts "Entrada invalida"
  puts "Tente novamente em outro momento"
end

puts
print "Tecle <Enter> para encerrar... "
gets
